### ISSUE TITLE
BUG: Datepicker UAT

### ISSUE THREAD

**Issue by emilygorelik** [ITS-HCD/nysds#1866]
1. [WCAG 2.4.3 – (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html) Focus moves from Today->Clear-><top row of date picker>, which is being flagged as not preserving the meaning of the content.
2. UAT comment - Hitting return after typing in the date does not close the date picker
3. UAT comment - Mobile users cannot type in the date
4. UAT comment - Focus does not automatically move between day, month, and year as you type on Safari

**Comment by mendeza** [comment]
> 1. [WCAG 2.4.3 – (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html) Focus moves from Today->Clear->, which is being flagged as not preserving the meaning of the content.

The graphical calendar is a popover. As a dialog-like element, it has focus contained, which descends in focus order, then wraps around to the top of the popover. This is fine.

---
CC @NovaCat35

**Comment by mendeza** [comment]
> 4\. UAT comment - Focus does not automatically move between day, month, and year as you type on Safari

That is a correct observation. The native HTML `<input type="date">` is used here. That is the experience the Safari rendering engine provides for the native element. It is different from Chrome; this does not make it wrong. Even in 2026 browsers still have differences. The upshot is actually good. A person who uses Safari as their main browser will find that the DS component behaves in this respect exactly like the multitude of sites that use a vanilla HTML date picker. That consistency is a good thing for them.

Compare Chrome vs Safari:
https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input/date

---
CC @NovaCat35

**Comment by mendeza** [comment]
> 3\. UAT comment - Mobile users cannot type in the date

- Under iOS and Android the native calendar UIs are presented to the user, so we rely on the respective native platforms' functionality. Question for the reporter: is this a limitation on what is expected for normal, idiomatic use?

---
CC @NovaCat35

**Comment by NovaCat35** [comment]
Thank you all for your contribution!
After much discussion, this ticket will be split into 2 smaller bug tickets, and this ticket will be closed.

**Tracking new tickets** 
- https://github.com/ITS-HCD/nysds/issues/1900 🎟️
- https://github.com/ITS-HCD/nysds/issues/1901 🎟️
